pkg/wakaama/contrib/lwm2m_objects/device/device.c

```diff
@@ -0,0 +1,357 @@
+                                         *data_arrayP + i);
+                result = COAP_205_CONTENT;
+                break;
+            case LWM2M_RES_MODEL_NO:
+                lwm2m_data_encode_string(LWM2M_DEVICE_MODEL, *data_arrayP + i);
+                result = COAP_205_CONTENT;
+                break;
+            case LWM2M_RES_SERIAL:
+                lwm2m_data_encode_string(LWM2M_DEVICE_SERIAL, *data_arrayP + i);
+                result = COAP_205_CONTENT;
+                break;
+            case LWM2M_RES_FW_VER:
+                lwm2m_data_encode_string(LWM2M_DEVICE_FW_VERSION, *data_arrayP + i);
+                result = COAP_205_CONTENT;
+                break;
+            case LWM2M_RES_HW_VERSION:
+                lwm2m_data_encode_string(LWM2M_DEVICE_HW_VERSION, *data_arrayP + i);
+                result = COAP_205_CONTENT;
+                break;
+            case LWM2M_RES_SW_VERSION:
+                lwm2m_data_encode_string(LWM2M_DEVICE_SW_VERSION, *data_arrayP + i);
+                result = COAP_205_CONTENT;
+                break;
+            case LWM2M_RES_BINDINGS:
+                lwm2m_data_encode_string(LWM2M_DEVICE_BINDINGS, *data_arrayP + i);
+                result = COAP_205_CONTENT;
+                break;
+            case LWM2M_RES_TYPE:
+                lwm2m_data_encode_string(LWM2M_DEVICE_TYPE, *data_arrayP + i);
+                result = COAP_205_CONTENT;
+                break;
+            /* Error code is mandatory */
+            case LWM2M_RES_ERROR_CODE:
+                /* TODO: Do this properly */
+                lwm2m_data_encode_int(0, *data_arrayP + i);
+                result = COAP_205_CONTENT;
+                break;
+            /* Time resources */
+            case LWM2M_RES_TIME:
+            case LWM2M_RES_TIME_OFFSET:
+            case LWM2M_RES_TIME_ZONE:
+            /* What to do with this? */
+            case LWM2M_RES_POWER_SRC:
+            case LWM2M_RES_POWER_VOL:
+            case LWM2M_RES_POWER_AMP:
+            case LWM2M_RES_BATTERY_LEVEL:
+            case LWM2M_RES_MEM_FREE:
+            case LWM2M_RES_BATTERY_STATUS:
+            case LWM2M_RES_MEM_TOTAL:
+            case LWM2M_RES_EXT_DEV_INFO:
+            default:
+                result = COAP_404_NOT_FOUND;
+        }
+
+        if (result != COAP_205_CONTENT) {
+            break;
+        }
+    }
+
+    return result;
+}
+
+static uint8_t prv_device_write(uint16_t instance_id, int num_data,
+                                lwm2m_data_t *data_array,
+                                lwm2m_object_t *objectP)
+{
+    int i;
+    uint8_t result = COAP_404_NOT_FOUND;
+    dev_data_t *data = (dev_data_t *)objectP->userData;
+
+    (void)data;
+
+    (void)instance_id;
+    (void)num_data;
+    (void)data_array;
+
+    for (i = 0; i < num_data; i++) {
+        switch (data_array[i].id) {
+            /* Writable */
+            case LWM2M_RES_TIME:
```

Comment:
Same thing here, is there a more sensible way to write this that does not involve listing everything? Is it possible to just check a range?

Reply:
I could add a 'count' value to the enum and check...